### PR TITLE
[WIP] Add minhash dedup job

### PR DESCRIPTION
**Pull request by Taekyoon**
### Features

- Add Minhash LSH Dedup Job
##### Job Parameters
- num_expand: number of generated minhash key
- sim_threshold: jaccard score threshold for text similarity
- n_gram: number of words(chars) to generate minhash key
- char_level: use n_gram as character level
- seed: set random seed to generate minhash 

#### How to test?

- test case
```json
{"text":"영화 완전 웃기지 않았어 ㅋㅋㅋㅋㅋㅋㅋㅋ"}
{"text":"영화 완전 웃기지 않았냐 ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ"}
{"text":"영화 완죤 웃기지 않았냐 ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ"}
{"text":"내 전화 번호는 022 - 9298 - 1113 이야"}
{"text":"아         오늘 저녁 맛있는거 먹고    싶    어. "}
{"text":"내 주민등록번호는 141555-4124492이야"}
{"text":"내 전화 번호는 022 - 9298 - 1213 이야"}
{"text":"<br> 안녕하세요 좋은 하루 되세요.!"}
{"text":"안녕하세요 좋은 하루 되세요.!<br>"}
{"text":"<br> 안녕하세요 좋은 하루 되세요.!<br>"}
{"text":"아         오늘 저녁 맛있는거 먹고    싶    다. "}
{"text":"제 카드 번호는 412 3456 789012 에요."}
{"text":"제 카드 번호는 412 2 - 3452 - 78902 에요."}
{"text":"제 계좌 번호는 412 - 345 - 678902 신한이 에요."}
```

- test command
```bash
python bin/sparkapp.py minhash_deduplication --input_dir {project_dir}/{your_test_file_dir} \
--output_dir ./test_outputs --n_gram 5 --num_expand 10 \
--sim_threshold 0.5 --char_level true --n_output 1
```

- test ouptuts
```python
{'text': '제 카드 번호는 412 2 - 3452 - 78902 에요.'},
{'text': '제 계좌 번호는 412 - 345 - 678902 신한이 에요.'},
{'text': '제 카드 번호는 412 3456 789012 에요.'},
{'text': '내 주민등록번호는 141555-4124492이야'},
{'text': '아         오늘 저녁 맛있는거 먹고    싶    어. '},
{'text': '영화 완전 웃기지 않았어 ㅋㅋㅋㅋㅋㅋㅋㅋ'},
{'text': '영화 완전 웃기지 않았냐 ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ'},
{'text': '내 전화 번호는 022 - 9298 - 1113 이야'},
{'text': '<br> 안녕하세요 좋은 하루 되세요.!'}
```

### CODE REVIEW

File: dps/spark/jobs/minhash_dedup_data.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            if group_represent_text in dedup_text:
+                yield dedup_text[0] if dedup_text[0] != group_represent_text else dedup_text[1]
+            else:
+                yield random.choice(dedup_text)
+
+
+def minhash_deduplication(input_dir: str, output_dir: str, 
+                          n_dist: int = 10, n_output: int = 10, 
+                          num_expand: int = 10, sim_threshold: float = 0.8, 
+                          n_gram: int = 15, char_level: bool = False, seed: int = 1):
```
Comment (hyunwoongko) [inline dps/spark/jobs/minhash_dedup_data.py:43]: how about using 13 as default? GPT3 used 13.

**Comment by hyunwoongko** [comment]
How about making a simple python test script for other teams?

File: dps/spark/jobs/minhash_dedup_data.py (hunk 1 above)
Comment (Taekyoon) [reply dps/spark/jobs/minhash_dedup_data.py:43]: Okay, I'll set 13 to `n_gram` as a default value :)
And, this parameter can be injected by command, so anyone can change this parameter setting.

File: dps/spark/jobs/minhash_dedup_data.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    with spark_session(f'') as spark:
+        sc: SparkContext = spark.sparkContext
+
+        proc_rdd: RDD = sc.textFile(input_dir) \
+            .repartition(n_dist) \
+            .flatMap(read_line) \
+            .cache()
+
+        overlap_kv_rdd: RDD = proc_rdd.flatMap(lambda x: expand_instances_by_minhash(x,
```
Comment (Fujiki-Nakamura) [inline dps/spark/jobs/minhash_dedup_data.py:53]: seems to be the same thing, but what about use `functools.partial`?
Just a suggestion but I mean something like
```python
from functools import partial
```
```python
_expand_instances_by_minhash = partial(expand_instances_by_minhash, expand_size=num_expand, n_gram=n_gram, seed=seed, char_level=char_level)
```
```python
overlap_kv_rdd: RDD = proc_rdd.flatMap(_expand_instances_by_minhash)\
```

Comment (Taekyoon) [reply dps/spark/jobs/minhash_dedup_data.py:53]: Sounds good! I can apply your suggestion :)

**Comment by Taekyoon** [comment]
> How about making a simple python test script for other teams?

Do you mean tests for functions or a job?

**Comment by hyunwoongko** [comment]
| Do you mean tests for functions or a job?

I mean functions.

**Pull request merged**